File: app/app/src/main/java/com/blog/app/entity/Article.java
package com.blog.app.entity;

import java.util.List;

public class Article {
    private Long id;
    private Long userId;
    private String title;
    private String contentMd;
    private String contentImg; // 封面图片URL
    private String contentMemo;
    private String articleType;
    private String articleLabel;
    private int articleStatus;
    private int browseCount;
    private int likeCount;
    private String createTime;
    private String updateTime;

    private List<ArticleType> articleTypes;

    // Getters and setters

    public static class ArticleType {
        private int id;
        private int parentId;
        private String typeName;
        private int num;
        private int node;
        private int createUser;
        private String createTime;
        private String updateTime;

        public int getId() {
            return id;
        }

        public void setId(int id) {
            this.id = id;
        }

        public int getParentId() {
            return parentId;
        }

        public void setParentId(int parentId) {
            this.parentId = parentId;
        }

        public String getTypeName() {
            return typeName;
        }

        public void setTypeName(String typeName) {
            this.typeName = typeName;
        }

        public int getNum() {
            return num;
        }

        public void setNum(int num) {
            this.num = num;
        }

        public int getNode() {
            return node;
        }

        public void setNode(int node) {
            this.node = node;
        }

        public int getCreateUser() {
            return createUser;
        }

        public void setCreateUser(int createUser) {
            this.createUser = createUser;
        }

        public String getCreateTime() {
            return createTime;
        }

        public void setCreateTime(String createTime) {
            this.createTime = createTime;
        }

        public String getUpdateTime() {
            return updateTime;
        }

        public void setUpdateTime(String updateTime) {
            this.updateTime = updateTime;
        }

        // Getters and setters
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getContentMd() {
        return contentMd;
    }

    public void setContentMd(String contentMd) {
        this.contentMd = contentMd;
    }

    public String getContentImg() {
        return contentImg;
    }

    public void setContentImg(String contentImg) {
        this.contentImg = contentImg;
    }

    public String getContentMemo() {
        return contentMemo;
    }

    public void setContentMemo(String contentMemo) {
        this.contentMemo = contentMemo;
    }

    public String getArticleType() {
        return articleType;
    }

    public void setArticleType(String articleType) {
        this.articleType = articleType;
    }

    public String getArticleLabel() {
        return articleLabel;
    }

    public void setArticleLabel(String articleLabel) {
        this.articleLabel = articleLabel;
    }

    public int getArticleStatus() {
        return articleStatus;
    }

    public void setArticleStatus(int articleStatus) {
        this.articleStatus = articleStatus;
    }

    public int getBrowseCount() {
        return browseCount;
    }

    public void setBrowseCount(int browseCount) {
        this.browseCount = browseCount;
    }

    public int getLikeCount() {
        return likeCount;
    }

    public void setLikeCount(int likeCount) {
        this.likeCount = likeCount;
    }

    public String getCreateTime() {
        return createTime;
    }

    public void setCreateTime(String createTime) {
        this.createTime = createTime;
    }

    public String getUpdateTime() {
        return updateTime;
    }

    public void setUpdateTime(String updateTime) {
        this.updateTime = updateTime;
    }

    public List<ArticleType> getArticleTypes() {
        return articleTypes;
    }

    public void setArticleTypes(List<ArticleType> articleTypes) {
        this.articleTypes = articleTypes;
    }
}

File: app/app/src/main/java/com/blog/app/data/model/article/Article.kt
package com.blog.app.data.model.article

/**
 * 博客内容服务返回的文章模型。
 */
data class Article(
    val id: Long,
    val userId: Long,
    val title: String,
    val contentMd: String,
    val contentImg: String?,
    val contentMemo: String,
    val articleFile: String?,
    val articleType: String,
    val articleLabel: String,
    val articleStatus: Int,
    val browseCount: Long,
    val likeCount: Long,
    val createTime: String,
    val updateTime: String,
    val articleTypes: List<ArticleTypeSummary>,
    val articleLabels: List<ArticleLabel>
) {
    /**
     * 提供给界面使用的 Markdown 正文。
     */
    val content: String
        get() = contentMd

    /**
     * 将文章分类名称组合为旧界面使用的显示文本。
     */
    val typeName: String
        get() = articleTypes.joinToString(" · ") { it.typeName }

    /**
     * 当前文章接口的列表数据没有提供作者名称。
     */
    val authorName: String
        get() = ""
}

/**
 * 文章响应中包含的分类信息。
 */
data class ArticleTypeSummary(
    val id: Long,
    val parentId: Long,
    val typeName: String
)

/**
 * 文章响应中包含的标签信息。
 */
data class ArticleLabel(
    val id: Long,
    val userId: Long,
    val labelType: Int,
    val labelName: String,
    val articleNum: Int
)
